### ISSUE TITLE
soroban-env-host: expose Token contract spec to CLI

### ISSUE THREAD

**Issue by 2opremio** [stellar/rs-soroban-env#535]
### What problem does your feature solve?

When invoking a contract, soroban-cli needs the contract's spec in order to parse the contract arguments.

With _normal_ contracts, we just obtain the contract's wasm from the network and extract the spec prior to parsing.

However, that's not possible for the token contract, which is implicit (the token contract's `ContractCode` ledger entry contains a `ScContractCode::Token` and not a `ScContractCode::Wasm(bytes)` ).

For now we have pushed the responsibility of passing the token contract's wasm to the end user, which is a hack.  See https://github.com/stellar/soroban-cli/pull/212

### What would you like to see?

I would like `soroban-env-host` (or any other library if more appropriate) to expose the Token contract spec for soroban-cli to use.

### What alternatives are there?

The token contract's code could be treated normally, so that its  `ContractCode` ledger entry contains wasm code, just like any other contract.

**Comment by leighmcculloch** [comment]
This is a duplicate of https://github.com/stellar/soroban-cli/issues/172.

I also don't think we should do this in the soroban-env-host, because the host has no knowledge of contract specs.

The `soroban-token-spec` in the rs-sorobans-sdk repo already exports the token spec.

The naive approach would be to take the .wasm file from that repo and embed it into the CLI, however doing so would be error prone, as we have already seen when we've used the same approach elsewhere. So I'd like to not make that same mistake we've made the previous times we've tried to solve this type of problem.

The approach I'd like to use is described briefly https://github.com/stellar/soroban-cli/issues/172, however it may not be safe to import the `soroban-token-spec` directly into the CLI due to how Rust merges static libs.
